data_safe_haven/infrastructure/programs/sre/workspaces.py

```diff
@@ -131,9 +132,13 @@ def __init__(
             ldap_user_filter=props.ldap_user_filter,
             ldap_user_search_base=props.ldap_user_search_base,
             software_repository_hostname=props.software_repository_hostname,
+            storage_account_data_desired_state_name=props.storage_account_data_desired_state_name,
             storage_account_data_private_user_name=props.storage_account_data_private_user_name,
             storage_account_data_private_sensitive_name=props.storage_account_data_private_sensitive_name,
-        ).apply(lambda kwargs: self.read_cloudinit(**kwargs))
+        ).apply(lambda kwargs: self.template_cloudinit(**kwargs))
+        b64cloudinit = Output.from_input(cloudinit).apply(
+            lambda cloudinit: b64encode(cloudinit)
+        )
```

Comment:
You shouldn't need `Output.from_input` as cloudinit should be an `Output[str]` already.

```suggestion
        b64cloudinit = cloudinit.apply(b64encode)
```